etcdserver/api/membership/cluster.go

```diff
@@ -759,16 +759,21 @@ func ValidateClusterAndAssignIDs(lg *zap.Logger, local *RaftCluster, existing *R
 	if len(ems) != len(lms) {
 		return fmt.Errorf("member count is unequal")
 	}
-	sort.Sort(MembersByPeerURLs(ems))
-	sort.Sort(MembersByPeerURLs(lms))
 
 	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
```

Comment:
since there are potentially more comparisons now, should the timeout value of 30 seconds be reconsidered? or is it better to move this inside the inner loop?

Reply:
There's a maximum of 49 resolutions, and 42 of those should be cached by the local resolver. The minimum possible DNS TTL is 1 second, and that's plenty of time to finish this loop, although more likely TTL is much higher anyway.

I think the 30 second timeout is only relevant to not blocking indefinitely if DNS is unresponsive